lib/linguist/languages.yml

```diff
@@ -412,6 +412,7 @@ Assembly:
   - nasm
   extensions:
   - ".asm"
+  - ".a86"
   - ".a51"
```

Comment:
Linguist requires extension lists to be sorted case-sensitively (sans the first entry, which is considered the “primary” extension), which means `.a86` should follow `.a51`, not precede it.

~~~diff
   extensions:
   - ".asm"
   - ".a51"
+  - ".a86"
~~~

Reply:
I didn't see that, sorry. Will change it later. However, afaik, .A51 (Intel 8051 I think) is not the same "language" as x86 assembly (8086, i286, i386 and co.). I see that Motorola 68K has its own thing, so maybe Intel 8051 should also have its own definition/whatever you call it? I can clearly see that GitHub is trying to apply the x86 syntax highlighting to the sample .A51 file. Maybe I should open an issue for that?

Again, sorry for not reading the contribution guidelines properly before opening this PR.

Reply:
> I see that Motorola 68K has its own thing

`Motorola 68K` is [grouped](https://github.com/github-linguist/linguist/blob/4ac734c15a96f9e16fd12330d0cb8de82274f700/lib/linguist/languages.yml#L30-L31) under `Assembly`, which means M68K falls under the Assembly umbrella for the purposes of language classification.

https://github.com/github-linguist/linguist/blob/4ac734c15a96f9e16fd12330d0cb8de82274f700/lib/linguist/languages.yml#L4385-L4399

The reason for making this distinction in the first place is to employ more accurate syntax highlighting (note the `tm_scope` field above). E.g., XML property lists use a specialised grammar for highlighting the [specialised XML subset](https://en.wikipedia.org/wiki/Property_list#Format) used by [<samp>plist(5)</samp>](https://www.unix.com/man-page/OSX/5/plist), even though there's nothing wrong with XML's usual grammar.

[^1]: `tm_scope` refers to the [<samp>scopeName</samp> field](https://macromates.com/manual/en/language_grammars#example_grammar) of TextMate-compatible grammars, which acts as a GUID of sorts.